### ISSUE TITLE
Mpileup finishes prematurely

### ISSUE THREAD

**Issue by rernst** [lomereiter/sambamba#153]
Hi Artem,

I am not sure whether this already happened in earlier versions of sambamba, but the latest release (v0.5.5) of sambamba mpileup finishes prematurely when generating pileups for whole exome data. 

I used the following command line:
```bash
sambamba_v0.5.5/sambamba mpileup -t 2 --tmpdir=/path/to/tmp -L exome.bed -o exome_sample.bam.pileup exome_sample.bam --samtools "-q 1 -f genome.fasta"
```

```bash
tail exome_sample.bam.pileup
1       24434446        C       31      ........,....,.,,,,..........,^].       Fkk7FkkkFFkkkFfFFFFFFkFFFFkfkAA
1       24434447        T       31      ........,....,.,,,,..........,. FkaAFk\\AFkakFkFFAAFFkFFFFfkSFA
1       24434448        C       31      ........,....,.,,,,..........,. FkfFFkkaFFkafFkFFFAFFkjFFFkkkFA
1       24434449        A       33      ........,....,.,,,,..........,.^].^].   .Nk<FNfkFFkkkFkFFFFAAkkA.FNkNA7A`
1       24434450        G       33      ........,....,.,,,,..........,...       FkfFjkkaFFkSkFkFFF.FFWkFFFkkkFAAf
1       24434451        C       33      ........,....,.,,,,..........,...       <kk7kk\\FFkkkFkFFFFFFkkF<FfkkFFAf
1       24434452        C       33      .......,....,.,,,..........,...^].^].   7kkkkfSFFkkkFkFFFFFkkFFFkkkAFAfAA
1       24434453        A       32      .......,..,.,,,.........,,.....^].      7S;a?kaFkkFaFFF.SS.7F\SkF>FAaAA<
1       24434454        C       36      ........,....,.,,,,..........,......    <kkFakkIFFkkkFkFFFFFAkkFFFkkfFFFkAAA
1       24434455        G       35      .....
```
Second sambamba run:
```bash
tail exome_sample.bam.pileup
1       46659945        C       33      ,.....,.,.,..,....,......,.,,....       FNakSfFFFk.kkFFFaF.\kFFSkFFF7kkFF
1       46659946        T       33      ,.....,.$,.,..,....,......,.,,....      FSNf;fA7Fa.akF.<SF7kaFFNk<FFFkkFF
1       46659947        G       32      ,.....,,.,..,....,......,.,,....        FIakWkFFkFkkAFAfF.kkFFakFFF7kfFF
1       46659948        G       31      ,.....,,.,..,..........,.,,.... Akfa6kF<k<kfAFFkFkkF<WkFFFFkfFF
1       46659949        G       33      ,.....,,.,..,....,......,.,,....^],     ASkkkkFFkFkkFFFkF.fkFFkk<F.FkkFFF
1       46659950        T       30      ,.....,,.,.............,,....,  .S6aNS..SAkNFFSFfkFFSS7FA>\FFF
1       46659951        C       32      ,.....,,.,..,..........,.,,....,        AfWkNfFFkFkkFFFkFkkFFkk7F.AkS\F.
1       46659952        A       33      ,$.....,,.,..,..........,.,,....,^].    Akfk>\FFk.k\FFFkAkkFFkkAFFFkSaFFA
1       46659953        G       33      .....,,.,..,....,......,.,,....,.       aS\NkFFkFkkFSFkFFkkFFfkFFFFkakFFA
1       46659954        G       31      .....,...,....,......,.,,....,. akk;SFfkkF
```
Bed files generated by sambamba during second run (in tmp folder):
```bash
==> 1.bed <==
1       68981   12164558
==> 2.bed <==
1       12164558        24434460
==> 3.bed <==
1       24434460        36552497
==> 4.bed <==
1       36552497        46660019
==> 5.bed <==
1       46660019        74648328
==> 6.bed <==
1       74648328        109715045
```

Using samtools (still running at the moment, but I guess you get the point.)
```bash
tail exome_sample.bam.pileup_samtools
4       998121  C       18      .,,,,.,....,.,..,,      kFAFFkFkkkFFFFFfAF
4       998122  C       18      .,,,,.,....,,...,,      aFAFFkFk;\FFFF7]AF
4       998123  C       19      .,,,,.,....,.,...,,     k<AFFkFkkkFFFAFFaFF
4       998124  T       17      .,,$,,....,,,...,,      fFAFFkakA:FF<.fFF
4       998125  G       18      .,,,.,....,.,...,,      aAAFkFfSfFF<FjFkFF
4       998126  T       18      .$,,,.,....,.,...,,     f<.FaA\ffFFAFfFIF<
4       998127  G       17      ,,,.,....,.,...,,       FFFfFkkk<FAFfFkF<
4       998128  C       18      ,,,,.,....,.,...,,      AFFFkFkkkFFAFkFkFA
4       998129  C       17      ,,,.,....,.,...,,       F<FkFk\kFFFFkFkF<
```

Please let me know if you need data/help to reproduce the error.

Robert

**Comment by lomereiter** [comment]
Hi Robert, please send me the data.

(By the way, the BED files in temporary directory were generated incorrectly, the input BED wasn't used - fixed that)

**Comment by rernst** [comment]
Confirmed fixed in: 3979e2808d2ba791fb2585d77731d436a0d45d8f
